### DISCUSSION TITLE
New installation is prompting to download the homepage as soon as i open it

### DISCUSSION THREAD

**Discussion by exico91** [kimai/kimai#3727] [On-Premise: Installation & Updates]
I'm installing kimai in a docker in Unraid, Im not using the official docker because for some reason i cannot open any webpage at the port in the docs.

Anyway, what i did is setup a nginx container with php 8 and tha latest kimai stable 1.30.1 following the instructions here: https://www.kimai.org/documentation/installation.html

I had already a container with mariadb and i created a db for kimai.

I edited the .env file and installed kimai. I can see tables in the new db but i get an error
"php.CRITICAL: Fatal Error: Allowed memory size of 134217728 bytes exhausted (tried to allocate 36864 bytes)"
I searched in the github issues and it points to file permissions but in the docker everything run as root and everything is writable so Im sure that its not a permissions issue (or at least i believe so). I also tried to set them to 777 but nothing changes.

Anyway now when i open the webpage it redirects to /en/homepage but it prompts me to download a homepage file in any browser i use.

Any suggestion?

---------------------------------
Solution:
If you are using a docker with nginx that has php support
Edit your nginx config with the one from a fresh installation and modify the settings accordingly. Dont just copy-paste.
https://www.kimai.org/documentation/fresh-ubuntu-20.html
and re-apply permissions
Also edit your php.ini file (normally in /etc/php8/php.ini or your ovverride file, in my case there is an ovverride in /config/php/php-local.ini

**Comment by kevinpapst** [comment]
> Allowed memory size of 134217728 bytes exhausted

Increase the PHP memory limit. 
This is not a Kimai issue but PHP hosting.
Please search online how to increase it, depending on your docker base image there might be different options and/or files to touch.

**Comment by exico91** [comment]
I've got past the error limit, I reinstalled everything and changed the memory limit from 128 MB to 512

Unfortunately it still prompts me to download the homepage as soon as i visit the web server and there is no log file in var/log , just the .gitkeep file

Im using this docker: https://github.com/linuxserver/docker-nginx


Anyway if this docker does not work, I can try again with the kimai official docker but i cannot get it to work. I mean, i can install it but there is nothing at the port 8001, it keeps erroring out at connection timeout. I've tried to setup the docker connection as hosted and also with its own IP but nothing.
In the docker console i can add a user (the super admin as instructed in the wiki)

**Reply by kevinpapst** [reply]
If you get the PHP file offered as download, then nginx is simply serving the file, instead of passing it to the PHP engine. 
But I am not a docker user and cannot support you with that.

Ask in the repository for help: https://github.com/tobybatch/kimai2

**Reply by exico91** [reply]
I made some progress, I followed this guide https://www.kimai.org/documentation/fresh-ubuntu-20.html for the configuration of nginx only.
I had already setup a docker with mariadb that works.
From the nginx perspective, the only things that i did not change are:
fastcgi_pass unix:/run/php/php8.1-fpm.sock;
I used 127.0.0.1:9000 instead, in this docker there is no sock from what i see
and
include fastcgi.conf;
because i do not have this file there, this docker include its own
include /etc/nginx/fastcgi_params;

Complete config:
```## Version 2022/10/04 - Changelog: https://github.com/linuxserver/docker-baseimage-alpine-nginx/commits/master/root/defaults/nginx/site-confs/default.conf.sample

server {
    listen 80 default_server;

#    listen 443 ssl http2 default_server;

    server_name _;

    access_log off;
    log_not_found off;
    root /config/www/kimai/public;
    index index.php;

    location / {
        # enable for basic auth
        #auth_basic "Restricted";
        #auth_basic_user_file /config/nginx/.htpasswd;

        try_files $uri /index.php$is_args$args;
    }

    location ~ ^/index\.php(/|$) {
        fastcgi_split_path_info ^(.+\.php)(/.*)$;
        fastcgi_pass 127.0.0.1:9000;
        fastcgi_index index.php;
        include /etc/nginx/fastcgi_params;
        fastcgi_param PHP_ADMIN_VALUE "open_basedir=$document_root/..:/tmp/";
        internal;
    }

    # deny access to .htaccess/.htpasswd files
    location ~ /\.ht {
        deny all;
    }
}
```

Unfortunately, is not working yet, it doesnt prompt me to download at least.
Now it redirects to /it/login but with a 500 error and a Kimai, (roughly translated since it is in Italian) Time monitoring, unexpected error, try again. If the problem persists contact the administrator

**Reply by exico91** [reply]
Well, the last step was easier than i tought. Permissions, lovely permissions.

Fixed with 
chmod -R g+r .
chmod -R g+rw var/

**Reply by kevinpapst** [reply]
And now it works?

**Reply by exico91** [reply]
Yes, works
